packages/flutter/lib/src/material/material.dart

```diff
@@ -14,6 +14,35 @@ import 'theme.dart';
 /// Used by [InkHighlight] and [InkSplash], for example.
 typedef RectCallback = Rect Function();
 
+/// Interactive states that some of the Material component widgets can take on when
+/// receiving input from the user.
+///
+/// States are defined by https://material.io/design/interaction/states.html#usage.
+///
+/// Some Material component widgets track their current state in a `Set<MaterialState>`.
+///
+/// See also:
+///  * [MaterialStateColor], a color that takes a callback function that can
+///    return a different color depending on the state of the component that it
+///    is used in.
+enum MaterialState {
+  /// The state when the user drags their mouse cursor over the given component.
+  ///
+  /// See: https://material.io/design/interaction/states.html#hover.
+  hovered,
+  /// The state when the user navigates with the keyboard to a given component.
+  ///
+  /// This can also sometimes be triggered with a component is tapped. For example,
```

Comment:
..triggered when a component is tapped...

Reply:
Done.